### DISCUSSION TITLE
The future of the MQTT codec in Netty 5

### DISCUSSION THREAD

**Discussion by ppatierno** [netty/netty#13739] [General]
Hi all,
I know that Netty 5 is not landed yet but, by reading this article [here](https://netty.io/wiki/new-and-noteworthy-in-5.0.html) and checking the official Netty repository, it's pretty clear that a choice was made about moving some "rarely" used codecs to a different repository, the netty-contrib one.
I have no insights to understand the reasons but it's totally understandable if it comes to make Netty more sustainable and focusing on the core and not on a bunch of codecs for it (which quite often need specific expertise on the protocol to handle, i.e. mqtt). Anyway, it would be interesting to know why the "rarely" definition was used for those codecs. I would agree for something like stomp, but not sure about mqtt.
The mqtt codec is the one I am interested in ... so my question is: what's the roadmap for everything which is out of the core netty repo but it's in the netty-contrib? The mqtt codec, for example, doesn't have a release yet. Is it just because Netty 5 was not released yet or because there are no contributors for it?
What's warring me is that last commit was done last year. Is it actually "in sync" with bugs/issues/features fixed/added with the same codec in Netty 4.x or it's just out of sync?
All these concerns come to understand if it's safe to use an mqtt codec from Netty 4 today for a new project but then having the problem of losing mqtt support from Netty when moving to version 5.
Can anyone clarify what are the thoughts from the community about this?
Thanks!

**Comment by chrisvest** [comment]
A number of things come together around this decision. The code in these codecs is usually unchanged from version to version, aside from version bumps in the pom files. It is also difficult for the Netty maintainers to have expertise in all the different codecs. The contrib repos have been set up so that others with interest and expertise can maintain them - and even cut releases to Maven Central - with minimum involvement from the core Netty maintainers.

**Reply by ppatierno** [reply]
Thanks @chrisvest so it sounds to me you are looking for volunteers. If no one will volunteer to take a codec repo, it would be actually not released and not supported anymore with Netty 5 because core Netty maintainers won't do that. Is this assumption right?

**Reply by ppatierno** [reply]
Hey @chrisvest ... any updates on my latest question/assumption?

**Reply by chrisvest** [reply]
Yeah, it's released on its own independent schedule.
